ml_frameworks_bot/keras_io/retriever.py

```diff
@@ -72,27 +85,42 @@ def load_documents(self, num_workers: Optional[int] = None) -> List[Document]:
             repository_name,
             personal_access_token,
         )
-        input_files = get_all_file_paths(
-            os.path.join(self.repository_local_path, "examples"),
-            included_file_extensions=[".md"],
-        )
-        input_files += get_all_file_paths(
-            os.path.join(self.repository_local_path, "guides"),
-            included_file_extensions=[".md"],
-        )
-        input_files += get_all_file_paths(
-            os.path.join(self.repository_local_path, "templates"),
-            included_file_extensions=[".md"],
-        )
-        reader = SimpleDirectoryReader(input_files=input_files)
-        return reader.load_data(num_workers=num_workers, show_progress=True)
+        source_directory = os.path.join(self.repository_local_path, "sources")
+        if not os.path.exists(source_directory):
+            build_keras_io_sources(repository_local_path=self.repository_local_path)
+        input_files = []
+        for directory in included_directories:
+            input_files += get_all_file_paths(
```

Comment:
could we do this with [glob](https://docs.python.org/3/library/glob.html) instead ?

Reply:
I don't think `glob` recursively fetches all files in a directory. Given that in our case, the depth of the directory tree is variable, `glob` can be used.